### PR TITLE
feat: upgrade Ray, Pydantic & FastAPI

### PR DESCRIPTION
**Pull request by Barabazs**
## Summary

* Upgrade the versions of Ray, Pydantic and FastAPI.
* refactor the pydantic syntax

## Motivation

Ray finally supports Pydantic >2.5

I haven't bumped the Aineko version yet, but we should release this as a new minor version.

## Author Checklist

- [x] Changes are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [ ] Documentation is updated if necessary.
- [x] Status checks pass.
- [ ] If the version number is changed, it is updated in `pyproject.toml` , `aineko/__init__.py`, `aineko/templates/first_aineko_pipeline/{{cookiecutter.project_slug}}/pyproject.toml`.

## Reviewer Checklist

- [ ] Title is accurate and formatted appropriately.
- [ ] No unnecessary changes are introduced.
- [ ] Description motivates each change.
- [ ] Docs are updated if necessary.
- [ ] Code is pythonic and consistent with the rest of the codebase.
- [ ] The code is consistent with the [style guide](https://google.github.io/styleguide/pyguide.html).

### CODE REVIEW

**Review by mdigi14 — APPROVED** [review]
This looks good to me. I did a pass through the docs and I don't see any instances where we would have to refactor examples (ie from using `.dict()` to `.model_dump()`). Another reviewer should double check on doc changes we might want to make.

File: aineko/models/config_schema.py [hunk 1]
```diff
@@ -16,16 +16,16 @@ class Dataset(BaseModel):
...
 
             class_name: str = Field(..., alias="class")
-            node_params: Optional[dict]
-            node_settings: Optional[dict]
-            inputs: Optional[list]
-            outputs: Optional[list]
+            node_params: Optional[dict] = None
+            node_settings: Optional[dict] = None
+            inputs: Optional[list] = None
+            outputs: Optional[list] = None
```
Comment (ricrosales) [inline aineko/models/config_schema.py:28]: Trying to understand this change. What is the rational for adding defaults to None throughout? Isn't this a change in behavior? I thought that `Optional[str]` by itself did not specifically have a default. I didn't fully trace the usage, do we use a default anywhere?

Comment (Barabazs) [reply aineko/models/config_schema.py:28]: `Optional[str]` by itself is just a type hint and shouldn't actually set None as a default value. But it's recommended to add the default value: [Required, optional, and nullable fields](https://docs.pydantic.dev/latest/migration/#required-optional-and-nullable-fields)

Comment (ricrosales) [reply aineko/models/config_schema.py:28]: Ok understood

**Review by ricrosales — APPROVED** [review]

**Review by craycrayfish — APPROVED** [review]
Thanks for doing this

**Pull request merged**